### PR TITLE
Currency component can update its value given new `value`

### PR DESCRIPTION
**Pull request by juanca**
https://www.pivotaltracker.com/story/show/176471537

## PR upkeep checklist

- [x] Change log entry
- [x] Label(s)
- [x] Assignee(s)
- [x] Deployment URL: https://mavenlink.github.io/design-system/updatable-currency-field/
- [x] (When ready for review) Reviewer(s)
- [x] Green Bigmaven CI check
- [ ] DevQA on Bigmaven staging (e.g. does the work need to be imported in the internal design system?)

### CODE REVIEW

**Review by cincospenguinos — APPROVED** [review]
LGTM homie

File: src/components/custom-field-input-currency/custom-field-input-currency.jsx [hunk 1]
```diff
@@ -63,6 +65,10 @@ const CustomFieldInputCurrency = forwardRef(function CustomFieldInputCurrency(pr
     setIsFocused(true);
   }
 
+  function onChange() {
+    props.onChange({ target: ref.current })
```
Comment (juanca) [inline src/components/custom-field-input-currency/custom-field-input-currency.jsx:69]:
```suggestion
    props.onChange({ target: ref.current });
```

File: src/components/custom-field-input-currency/custom-field-input-currency.jsx [hunk 2]
```diff
@@ -150,7 +161,11 @@ CustomFieldInputCurrency.propTypes = {
   id: PropTypes.string.isRequired,
   label: PropTypes.string.isRequired,
   name: PropTypes.string.isRequired,
-  // onChange: Do not expose an onChange handler. See commit for details.
+  /**
+   * The handle is invoked for every native onchange event.
```
Comment (juanca) [inline src/components/custom-field-input-currency/custom-field-input-currency.jsx:165]:
```suggestion
   * The handler is invoked for every native onchange event.
```

File: src/components/custom-field-input-currency/custom-field-input-currency.jsx [hunk 3]
```diff
@@ -150,7 +161,11 @@ CustomFieldInputCurrency.propTypes = {
...
+   * The handle will be invoked with the forwarded ref.
```
Comment (juanca) [inline src/components/custom-field-input-currency/custom-field-input-currency.jsx:166]:
```suggestion
   * The handler will be invoked with the forwarded ref.
```

File: src/components/custom-field-input-number/custom-field-input-number.jsx [hunk 4]
```diff
@@ -57,7 +58,12 @@ CustomFieldInputNumber.propTypes = {
   label: PropTypes.string.isRequired,
   name: PropTypes.string,
   onBlur: PropTypes.func,
-  // onChange: Do not expose the onChange handler. See commit for details.
+  /**
+   * The handle is invoked for every native onchange event.
```
Comment (juanca) [inline src/components/custom-field-input-number/custom-field-input-number.jsx:62]:
```suggestion
   * The handler is invoked for every native onchange event.
```

File: src/components/number/number.jsx [hunk 5]
```diff
@@ -106,6 +106,11 @@ Number.propTypes = {
   label: PropTypes.string.isRequired,
   name: PropTypes.string,
   onBlur: PropTypes.func,
+  /**
+   * The handle is invoked for every native onchange event.
```
Comment (juanca) [inline src/components/number/number.jsx:110]:
```suggestion
   * The handler is invoked for every native onchange event.
```

**Comment by juanca** [comment]
Building @ https://github.com/mavenlink/mavenlink/pull/23068

**Review by cincospenguinos — APPROVED** [review]
LGTM once again! Glad we got `onChange` in place now

**Pull request merged**